File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    string s;

    // Prompt for user input longer than 1 character.
    do
    {
        s = get_string("Text: ");
    }
    while (strlen(s) < 1);

    // Check each character
    for (int i = 0, l = strlen(s); i < l; i++)
    {
        // Increment count if letter
        if (isalpha(s[i]))
        {
            letters++;
        }

        // Increment count if space
        if (isspace(s[i]))
        {
            words++;
        }

        // Increment count if end of sentence
        if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            sentences++;
        }
    }
    // Calculate amount of letters per 100 words.
    float letperhun = (float) letters / words * 100;

    // Calculate amount of sentences per 100 words.
    float senperhun = (float) sentences / words * 100;

    // Calculate grade and round number
    int grade = round(0.0588 * letperhun - 0.296 * senperhun - 15.8);

    // Print grade
    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade > 1)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: sentimental-readability/readability.py
# import cs50 library function get_string
from cs50 import get_string


def main():
    # Initialize variables
    letters = 0
    words = 1
    sentences = 0
    text = get_input()

    # Check for character occurences
    for char in text:
        if char.isalpha():
            letters += 1
        if char.isspace():
            words += 1
        if char == "." or char == "?" or char == "!":
            sentences += 1

    letperhun = letters / words * 100
    senperhun = sentences / words * 100

    # Calculate grade with Coleman-Liau index
    grade = round(0.0588 * letperhun - 0.296 * senperhun - 15.8)

    # Print result
    if grade > 16:
        print("Grade 16+")
    elif grade > 1:
        print("Grade {}".format(grade))
    else:
        print("Before Grade 1")


# Get user input function
def get_input():
    while True:
        s = get_string("Text: ")
        if s:
            return s


if __name__ == "__main__":
    main()